inc/gravity-forms.php

```diff
@@ -61,16 +61,51 @@ function ( $form_string, $form ) {
 		$form_string = str_replace( 'gform_save_link button', 'nhsuk-button nhsuk-button--secondary', $form_string );
 		$form_string = str_replace( 'gform_save_link', 'nhsuk-button nhsuk-button--secondary', $form_string );
 
-		$outerfind[]    = 'gfield_error';
-		$outerreplace[] = 'gfield_error nhsuk-form-group--error';
-		$form_string    = str_replace( $outerfind, $outerreplace, $form_string );
+		$form_string = add_nhsuk_group_error_class_for_validation( $form_string );
 		$form_string    = add_nhsuk_class( $form_string ); // Add  'nhsuk-form-group' to 'gfield'.
 		return $form_string;
 	},
 	10,
 	2
 );
 
+/**
+ * Adds nhsuk form group error class to gravity form div having individual form elemen(s)
+ * with gfield error class.
+ *
+ * @param mixed $html html content.
+ * @return bool|string
+ */
+function add_nhsuk_group_error_class_for_validation( $html ) {
+	$doc = new DOMDocument();
+	libxml_use_internal_errors( true ); // Suppress warnings from malformed HTML.
+	$doc->loadHTML( $html );
+	libxml_clear_errors();
+
+	$xpath      = new DOMXPath( $doc );
+	$outer_divs = $xpath->query( "//div[contains(@class, 'ginput_container')]" );
+
+	if ( $outer_divs->length > 0 ) {
+		foreach ( $outer_divs as $outer_div ) {
+			// Find the nested inner div with class 'gfield_error is-error' inside the outer div.
+			$inner_divs = $xpath->query( ".//*[contains(@class, 'gfield_error is-error')]", $outer_div );
+			if ( $inner_divs->length > 0 ) {
+				// Append the 'nhsuk-form-group--error' class to the outer div.
+				$existing_class = $outer_div->getAttribute( 'class' );
+				$new_class      = 'nhsuk-form-group--error';
+				$class_array    = explode( ' ', $existing_class );
+				if ( ! in_array( $new_class, $class_array, true ) ) {
```

Comment:
instead of crateing array it might be quicker to check 
```
if (false === strstr($exiting_class, $new_class)) {
   $updated_class = trim( $existing_class . ' ' . $new_class );
   $outer_div->setAttribute( 'class', $updated_class );
}
```